dbms/src/Core/SpillHandler.cpp

```diff
@@ -174,5 +192,10 @@ void SpillHandler::finish()
         current_spilled_file_index = INVALID_CURRENT_SPILLED_FILE_INDEX;
         RUNTIME_CHECK_MSG(spiller->isSpillFinished() == false, "{}: spill after the spiller is finished.", spiller->config.spill_id);
     }
+    else if unlikely (spiller->isAllConstant())
+    {
+        std::lock_guard lock(spiller->all_constant_mutex);
```

Comment:
I suggest to keep total_rows in local and commit it to spiller in `SpillHandler::finish()`, just like the normal block spill.

Reply:
ok, updated.